### DISCUSSION TITLE
How about writing data to databend via flink ？

### DISCUSSION THREAD

**Discussion by artorias1024** [databendlabs/databend#4853] [Ideas]
If I implement a flink connector, is it possible to bypass the compute layer and write data directly to storage, similar to hudi?

**Comment by xudong963** [comment]
Hi @artorias1024, I think we can directly use `flink.streaming.api.functions.sink.RichSinkFunction`. 
I have done some small experiments before, flink->clickhouse, I think by this idea, flink->databend is also possible. https://github.com/xudong963/realtime-system/blob/master/src/main/java/org/data/dataflow/FlinkToCK.java. 
Please try it, feel free to ping me when you encounter problems!

**Reply by xudong963** [reply]
Hi @artorias1024 , how it's going？

**Reply by artorias1024** [reply]
My idea is to avoid building the data written to DataBend via SQL parsing, which would introduce some performance loss (in batch processing)? Is it possible to write directly to the storage layer by means of RPC directly?

**Reply by bohutang** [reply]
Makes sense, but databend there is no storage RPC yet, any ideas on the RPC API design? 
Thanks.

**Reply by artorias1024** [reply]
So far I have only a very rough idea that the API for the storage layer should be relatively simple. For example, the interface exposed to the public should be similar to 'Put', 'Get', 'Scan'. For Databend, these APIs need to get the metadata information, which can be obtained through Meta-Server. For Flink, different data sources are abstracted into different 'Sources', which are not only used for reading data, but also involved in Join for streaming computation.
